Clarify documentation for usage of TLS registry in Quarkus clients (#47716)
quarkusio/quarkus: docs/src/main/asciidoc/tls-registry-reference.adoc

For instance, `quarkus.tls.reload-period` will only be applied to the default TLS configuration.
====

[IMPORTANT]
====

> Due to link:https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc#configuring-clients-with-the-tls-registry[security considerations], Quarkus client extensions (REST, GRPC, etc) ignore properties, defined in default (ie unnamed) TLS configuration.

Reviewer: ```suggestion
As desribed in detail link:https://github.com/quarkusio/quarkus/blob/main/adr/0004-using-the-tls-registry-for-clients.adoc#configuring-clients-with-the-tls-registry[here], Quarkus client extensions (REST, GRPC, etc) ignore properties, defined in default (ie unnamed) TLS configuration.
```
